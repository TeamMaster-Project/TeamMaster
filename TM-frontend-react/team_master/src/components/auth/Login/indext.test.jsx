import React from 'react'
import {render, screen, cleanup } from '@testing-library/react';
import renderer from 'react-test-renderer';
import LoginForm from '.';


afterEach(()=>{
    cleanup();
})

// test("Should Render Todo Component", () => { 
//     // expect(true).toBe(true);
//     render(<LoginForm/>);
//     const todoElement = screen.getByTestId('todo-1');
//     expect(todoElement).toBeInTheDocument();
//     expect(todoElement).toHaveTextContent('Hello');
// })

test('Matches Snapshot',()=>{
    const tree = renderer.create(<LoginForm/>).toJSON();
    expect(tree).toMatchSnapshot();
})

describe('Testing UI', () => {
    it('Form elements rendering correctly', async () => {
        const { getByRole, getByTestId } = render(<LoginForm/>);

        const emailInput = getByTestId("username");
        expect(emailInput).toBeInTheDocument();

        const passwordInput = getByTestId("password");
        expect(passwordInput).toBeInTheDocument()

        expect(screen.getByRole('button', { name: 'Login' })).toBeInTheDocument();

        // expect(screen.getByText("Email / Username")).toBeInTheDocument();
    });  
});