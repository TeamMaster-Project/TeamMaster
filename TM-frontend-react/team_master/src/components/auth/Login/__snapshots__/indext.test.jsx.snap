// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Matches Snapshot 1`] = `
<div
  class="container-fluid"
>
  <div
    class="row no-gutter"
  >
    <div
      class="col-md-6 d-none d-md-flex bg-image"
    />
    <div
      class="col-md-6 bg-light"
    >
      <div
        class="login d-flex align-items-center py-5"
      >
        <div
          class="container"
        >
          <div
            class="row"
          >
            <div
              class="col-lg-10 col-xl-7 mx-auto"
            >
              <h3
                class="display-4"
              >
                TeamMaster
              </h3>
              <h3
                class="display-4 mb-5"
              >
                Log In
              </h3>
              <h6
                class="text-muted mb-4"
              >
                Enter your email and password
              </h6>
              <div
                className="register-form"
              >
                <form
                  onSubmit={[Function]}
                >
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="username"
                    >
                      Email / Username
                    </label>
                    <input
                      className="form-control"
                      data-testid="username"
                      id="username"
                      name="username"
                      onChange={[Function]}
                      type="text"
                      value=""
                    />
                  </div>
                  <div
                    className="form-group"
                  >
                    <label
                      htmlFor="password"
                    >
                      Password
                    </label>
                    <input
                      className="form-control"
                      data-testid="password"
                      id="password"
                      name="password"
                      onChange={[Function]}
                      type="password"
                      value=""
                    />
                  </div>
                  <div
                    className="form-submit-button"
                  >
                    <button
                      className="btn btn-primary"
                      data-testid="Login"
                      disabled={
                        Object {
                          "password": "\\"Password\\" is not allowed to be empty",
                          "username": "\\"Username\\" is not allowed to be empty",
                        }
                      }
                    >
                      Login
                    </button>
                  </div>
                </form>
              </div>
              <h6
                class="text-muted mb-4"
              >
                 Don't you have a TeamMaster account ? Â 
                <a
                  className="mb-4"
                  href="/register"
                  onClick={[Function]}
                >
                  <i>
                    Create New Project
                  </i>
                </a>
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
